numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2652,21 +2652,31 @@ einsum_list_to_subscripts(PyObject *obj, char *subscripts, int subsize)
         /* Subscript */
         else if (PyInt_Check(item) || PyLong_Check(item)) {
             long s = PyInt_AsLong(item);
-            if ( s < 0 || s > (2*26 - 1) ) {
+            npy_bool bad_input = 0;
+
+            if (subindex + 1 >= subsize) {
```

Comment:
Why the +1? Null terminator?

Reply:
I didn't think about it too much, I just saw a similar check just above it in the ellipsis. Come to think of it - I will change that back since there is no point to it.

Reply:
I took a look through - yeah, it appears that we need to leave some space for the terminator otherwise it will try to write it out of allocated memory.